sw/device/lib/dif/dif_rstmgr.c

```diff
@@ -0,0 +1,161 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/dif/dif_rstmgr.h"
+
+#include "sw/device/lib/base/bitfield.h"
+#include "sw/device/lib/base/mmio.h"
+#include "rstmgr_regs.h"  // Generated.
+
+// TODO - use bitfields library where appropriate.
+// TODO - add regen status get function.
+
+static const bitfield_field32_t kRstmgrCausesField = {
+    .mask = RSTMGR_RESET_INFO_CAUSE_MASK,
+    .index = RSTMGR_RESET_INFO_CAUSE_OFFSET,
+};
+
+// Regen registers are "write 1 to clear".
+static const ptrdiff_t kRstmgrRegenOffsetsMap[kDifRstmgrPeripheralLast + 1] = {
```

Comment:
Note: this and the next `*Map` arrays might become obsolete if `regen` and `reset` control bits are packed in a single `regen` and `reset` registers, instead of the current one register per peripheral. Please see #3089 for further information.

Reply:
Marking this as solved.